https://gitlab.freedesktop.org/xorg/driver/xf86-video-vmware/-/commit/50ab008cb965dc9e47e0a84a1950e2485bbd3c44
https://bugs.gentoo.org/829627

From: Martin Krastev <krastevm@vmware.com>
Date: Tue, 30 Nov 2021 11:47:39 -0800
Subject: [PATCH] vmwgfx: Change header inclusion order to avoid xorg headers
 catching stdbool.h

libdrm commit e641e2a632d779f638ac2ba983b9fceb20b3fac4 added
stdbool.h to the library headers which conflicts with xorg headers.

Signed-off-by: Martin Krastev <krastevm@vmware.com>
Reviewed-by: Zack Rusin <zackr@vmware.com>
Signed-off-by: Zack Rusin <zackr@vmware.com>
--- a/vmwgfx/vmwgfx_driver.h
+++ b/vmwgfx/vmwgfx_driver.h
@@ -36,15 +36,15 @@
 #include <stddef.h>
 #include <stdint.h>
 #include <errno.h>
-#include <drm.h>
-#include <xf86drm.h>
-#include <xf86drmMode.h>
 #include <xorg-server.h>
 #include <xf86.h>
 #include <xf86Crtc.h>
 #include <xf86xv.h>
 #include <xa_tracker.h>
 #include <xf86Module.h>
+#include <drm.h>
+#include <xf86drm.h>
+#include <xf86drmMode.h>
 
 #include "../src/compat-api.h"
 #ifdef DRI2
--- a/vmwgfx/vmwgfx_saa.c
+++ b/vmwgfx/vmwgfx_saa.c
@@ -29,11 +29,11 @@
 #include <xorgVersion.h>
 #include <mi.h>
 #include <fb.h>
-#include <xf86drmMode.h>
 #include <xa_context.h>
 #include "vmwgfx_saa.h"
 #include "vmwgfx_drmi.h"
 #include "vmwgfx_saa_priv.h"
+#include <xf86drmMode.h>
 
 /*
  * Damage to be added as soon as we attach storage to the pixmap.
GitLab
